code/auth-service/src/permissions/permissions.js

```diff
@@ -6,39 +6,37 @@ const roleDelim = ':';
 
 const permissionAttributes = yaml.safeLoad(fs.readFileSync(config.get('permissionAttributes')));
 
-const parseRoles = role => role.split(roleDelim, 2);
-
-const parseProjects = splitRoles => ({
-  role: splitRoles.pop(),
-  projectName: splitRoles.pop(),
-});
-
+//    [ { role: 'admin', projectName: 'project' } ]
+// -> [ { role: 'admin', projectName: 'project', permissions: [stackAttributes, storageAttributes, userAttributes] } ]
 const getPermissions = roleObject => ({
   ...roleObject,
   permissions: permissionAttributes[roleObject.role] || [],
 });
 
-const flattenArray = (previous, current) => {
-  if (Array.isArray(current)) {
-    return [...previous, ...current];
-  }
-  return previous;
-};
-
+//    [ { role: 'admin', projectName: 'project', permissions: [stackAttributes, storageAttributes, userAttributes] } ]
+// -> [ { role: 'admin', projectName: 'project', permissions: [/see permissions.yml/] } ]
 const buildPermissions = ({ permissions, ...rest }) => ({
   ...rest,
   permissions: permissions.map(({ name, permissions: subPermissions }) => subPermissions.map(permission => name.concat(roleDelim, permission)))
     .reduce(flattenArray, []),
 });
 
+//    [ { role: 'admin', projectName: 'project', permissions: [/see permissions.yml/] } ]
+// -> [ ['project:stacks:delete', ...] ]
 const projectifyPermissions = ({ projectName, permissions }) => {
   const project = projectName ? `${projectName}${roleDelim}` : '';
 
   return permissions.map(permission => project.concat(permission));
 };
 
-const processRoles = roles => roles.map(parseRoles)
-  .map(parseProjects)
+const flattenArray = (previous, current) => {
```

Comment:
Given lodash is already a dependency we could probably just use `flatten` or `concat` from there rather than write our own flatten function..probably `concat`
https://lodash.com/docs/4.17.15#concat
https://lodash.com/docs/4.17.15#flatten

Reply:
This wasn't as simple as it seemed as the function passed to reduce takes 4 arguments meaning that replacement with flatten didn't work. Leaving as is.